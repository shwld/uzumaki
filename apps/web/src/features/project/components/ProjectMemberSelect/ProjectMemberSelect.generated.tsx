import * as Types from '../../../../graphql/generated/graphql';

import gql from 'graphql-tag';
import * as Urql from 'urql';
export type DateString = string & { __dateStringBrand: any };
export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export type ProjectMemberSelect_MemberFragment = {
  __typename?: 'ProjectMember';
  id: string;
  role: Types.ProjectMemberRole;
  name: string;
  isMe: boolean;
};

export type ProjectMemberSelectQueryVariables = Types.Exact<{
  projectId: Types.Scalars['ID'];
}>;

export type ProjectMemberSelectQuery = {
  __typename?: 'Query';
  viewer?:
    | {
        __typename?: 'Viewer';
        id: string;
        project?:
          | {
              __typename?: 'Project';
              id: string;
              members: {
                __typename?: 'ProjectMemberConnection';
                edges?:
                  | Array<
                      | {
                          __typename?: 'ProjectMemberEdge';
                          cursor?: string | undefined;
                          node?:
                            | {
                                __typename?: 'ProjectMember';
                                id: string;
                                role: Types.ProjectMemberRole;
                                name: string;
                                isMe: boolean;
                              }
                            | undefined;
                        }
                      | undefined
                    >
                  | undefined;
                pageInfo?:
                  | {
                      __typename?: 'PageInfo';
                      hasNextPage: boolean;
                      endCursor?: string | undefined;
                    }
                  | undefined;
              };
            }
          | undefined;
      }
    | undefined;
};

export const ProjectMemberSelect_MemberFragmentDoc = gql`
  fragment ProjectMemberSelect_Member on ProjectMember {
    id
    role
    name
    isMe
  }
`;
export const ProjectMemberSelectDocument = gql`
  query ProjectMemberSelect($projectId: ID!) {
    viewer {
      id
      project(id: $projectId) {
        id
        members {
          edges {
            node {
              ...ProjectMemberSelect_Member
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
        }
      }
    }
  }
  ${ProjectMemberSelect_MemberFragmentDoc}
`;

export function useProjectMemberSelectQuery(
  options: Omit<Urql.UseQueryArgs<ProjectMemberSelectQueryVariables>, 'query'>
) {
  return Urql.useQuery<
    ProjectMemberSelectQuery,
    ProjectMemberSelectQueryVariables
  >({ query: ProjectMemberSelectDocument, ...options });
}
