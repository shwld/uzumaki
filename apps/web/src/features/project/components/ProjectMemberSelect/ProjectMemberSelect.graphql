fragment ProjectMemberSelect_Member on ProjectMember {
  id
  role
  isMe
  profile {
    name
  }
}

query ProjectMemberSelect($projectId: ID!) {
  viewer {
    id
    project(id: $projectId) {
      id
      members {
        edges {
          node {
            ...ProjectMemberSelect_Member
          }
          cursor
        }
        pageInfo {
          hasNextPage
          endCursor
        }
      }
    }
  }
}
