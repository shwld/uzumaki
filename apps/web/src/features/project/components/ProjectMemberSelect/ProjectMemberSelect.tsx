import { Select, SelectProps, forwardRef } from '@chakra-ui/react';
import { useProjectMemberSelectQuery } from './ProjectMemberSelect.generated';

export const ProjectMemberSelect = forwardRef<
  { projectId: string } & SelectProps,
  'select'
>(({ projectId, ...props }, ref) => {
  const [{ data, fetching, error }] = useProjectMemberSelectQuery({
    variables: { projectId },
  });
  if (error != null) return <></>;
  if (fetching) return <></>;
  if (data?.viewer?.project == null) return <></>;

  return (
    <Select
      {...props}
      ref={ref}
      defaultValue={
        data.viewer.project.members.edges?.find(it => !!it?.node?.isMe)?.node
          ?.id
      }
    >
      {data.viewer.project.members.edges?.map(member => (
        <option key={member?.node?.id} value={member?.node?.id}>
          {member?.node?.name}
        </option>
      ))}
    </Select>
  );
});
