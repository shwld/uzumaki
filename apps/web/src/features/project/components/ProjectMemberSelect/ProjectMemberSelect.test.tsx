import { ProjectMemberSelect } from './ProjectMemberSelect';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';

describe('ProjectMemberSelect', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <ProjectMemberSelect projectId="test" />
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('sapiente')).toBeTruthy();
  });
});
