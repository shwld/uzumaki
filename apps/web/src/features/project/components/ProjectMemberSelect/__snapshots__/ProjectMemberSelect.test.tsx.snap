// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectMemberSelect Snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-select__wrapper css-42b2qy"
  >
    <select
      class="chakra-select css-1ik61og"
    >
      <option
        value="d9d74e52-e374-4a7a-94e3-ea15097779b8"
      >
        sapiente
      </option>
    </select>
    <div
      class="chakra-select__icon-wrapper css-162mkon"
    >
      <svg
        aria-hidden="true"
        class="chakra-select__icon"
        focusable="false"
        role="presentation"
        style="width: 1em; height: 1em; color: currentColor;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
